weave/artifact_wandb.py

```diff
@@ -908,3 +914,46 @@ def parse(cls, uri: str) -> "WeaveWBLoggedArtifactURI":
 
     def to_ref(self) -> WandbArtifactRef:
         return WandbArtifactRef.from_uri(self)
+
+
+# This is a wrapper around an artifact that acts like a list of files.
+# It fetchs a file from the manifest on __getItem__ and can return a count without fetching all files
+@dataclasses.dataclass
+class FilesystemArtifactFileIterator(list[artifact_fs.FilesystemArtifactFile]):
+    data: list[str]
+    artifact: WandbArtifact
+    idx: int = 0
+
+    @property
+    def type(self) -> types.Type:
```

Comment:
Can you share why this type property is needed?

Reply:
Oh i dont think it is, let me remove that